pycalphad/tests/test_database.py

```diff
@@ -684,4 +684,104 @@ def test_load_database_when_given_in_lowercase():
+    ELEMENT PB   FCC_A1                    2.0720E+02  6.8785E+03  6.4785E+01 !
+    ELEMENT PD   FCC_A1                    1.0642E+02  5.4685E+03  3.7823E+01 !
+    ELEMENT PR   DHCP                      1.4091E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT PT   FCC_A1                    1.9508E+02  5.7237E+03  4.1631E+01 !
+    ELEMENT PU   ALPHA_PU                  2.4406E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT RB   BCC_A2                    8.5468E+01  7.4894E+03  7.6776E+01 !
+    ELEMENT RE   HCP_A3                    1.8621E+02  5.3555E+03  3.6526E+01 !
+    ELEMENT RH   FCC_A1                    1.0291E+02  4.9204E+03  3.1505E+01 !
+    ELEMENT RU   HCP_A3                    1.0107E+02  4.6024E+03  2.8535E+01 !
+    ELEMENT S    ORTHORHOMBIC_S            3.2066E+01  0.0000E+00  0.0000E+00 !
+    ELEMENT SB   RHOMBOHEDRAL_A7           1.2175E+02  5.8702E+03  4.5522E+01 !
+    ELEMENT SC   HCP_A3                    4.4956E+01  0.0000E+00  0.0000E+00 !
+    ELEMENT SE   HEXAGONAL_A8              7.8960E+01  5.5145E+03  4.1966E+01 !
+    ELEMENT SI   DIAMOND_A4                2.8085E+01  3.2175E+03  1.8820E+01 !
+    ELEMENT SM   RHOMBOHEDRAL_SM           1.5036E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT SN   BCT_A5                    1.1871E+02  6.3220E+03  5.1195E+01 !
+    ELEMENT SR   FCC_A1                    8.7620E+01  0.0000E+00  0.0000E+00 !
+    ELEMENT TA   BCC_A2                    1.8095E+02  5.6819E+03  4.1472E+01 !
+    ELEMENT TB   HCP_A3                    1.5893E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT TC   HCP_A3                    9.7907E+01  0.0000E+00  0.0000E+00 !
+    ELEMENT TE   HEXAGONAL_A8              1.2760E+02  6.1212E+03  4.9497E+01 !
+    ELEMENT TH   FCC_A1                    2.3204E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT TI   HCP_A3                    4.7880E+01  4.8100E+03  3.0648E+01 !
+    ELEMENT TL   HCP_A3                    2.0438E+02  6.8283E+03  6.4183E+01 !
+    ELEMENT TM   HCP_A3                    1.6893E+02  7.3973E+03  7.4015E+01 !
+    ELEMENT U    ORTHORHOMBIC_A20          2.3803E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT V    BCC_A2                    5.0941E+01  4.5070E+03  3.0890E+01 !
+    ELEMENT W    BCC_A2                    1.8385E+02  4.9700E+03  3.2620E+01 !
+    ELEMENT Y    HCP_A3                    8.8906E+01  0.0000E+00  0.0000E+00 !
+    ELEMENT YB   FCC_A1                    1.7304E+02  0.0000E+00  0.0000E+00 !
+    ELEMENT ZN   HCP_A3                    6.5390E+01  5.6568E+03  4.1631E+01 !
+    ELEMENT ZR   HCP_A3                    9.1224E+01  5.5663E+03  3.9181E+01 !
+
+    PHASE LIQUID % 1 1 !
+    CONSTITUENT LIQUID : AG,AL,AM,AS,AU,B,BA,BE,BI,C,CA,CD,CE,CO,CR,CS,CU,DY,ER,
+    EU,FE,GA,GD,GE,HF,HG,HO,IN,IR,K,LA,LI,LU,MG,MN,MO,N,NA,NB,ND,NI,NP,O,OS,P,PA,
+    PB,PD,PR,PT,PU,RB,RE,RH,RU,S,SB,SC,SE,SI,SM,SN,SR,TA,TB,TC,TE,TH,TI,TL,TM,U,V,
+    W,Y,YB,ZN,ZR : !
+    """
+    dbf = Database(TDB)
```

Comment:
Can we assert that the expected number of components is in the Database prior to `to_string`? I'm imagining a case where a parse error inadvertently drops the constituents broken over subsequent lines, and this test incorrectly passes.

Reply:
Good idea. I added some more rigorous checks.